src/buildQuery.ts

```diff
@@ -38,7 +38,7 @@ let typeMap: any
 let queryMap: QueryMap
 
 const mapType = (idType: any, value: string | number) =>
-  idType.name === 'String' ? value : parseInt(value as string, 10)
+  idType.name === 'String' ? value : idType.name === 'UUID' ? value : parseInt(value as string, 10)
```

Comment:
```suggestion
  ['UUID', 'Uuid', 'String'].includes(idType.name) ? value : parseInt(value as string, 10)
```

:sparkles: legacy

(We used to use the lowercase form `Uuid`, and users can still opt into this, so just support both. Also simplified with this in mind.)

Reply:
Thank you! Makes sense :) We could also directly check the lowercased version's maybe instead:

`['uuid', 'string'].includes(idType.name.toLowerCase()) `

We check the type here: https://github.com/BowlingX/ra-postgraphile/blob/master/src/filters.ts#L18 again. So maybe it makes sense to introduce a normalize function for the names.